bin/.gitignore

```diff
@@ -1,2 +1,2 @@
 cache
-
+flutter.lock
```

Comment:
Ah, fair enough. In that case I would create a second file next to the first specifically for the batch file.

The point of having the file be on the filesystem at all times is to discourage people from thinking that the file's existence is relevant to whether things are locked. It should be trivially provable that everyone should always have the file, and thus the lock can't be based on the existence of the file.